diff --git a/setup.py b/setup.py
index 506f77b..5d72bc3 100755
--- a/setup.py
+++ b/setup.py
@@ -67,5 +67,5 @@ setup(
     extras_require=extras_require,
     python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*',
     classifiers=classifiers,
-    entry_points={'console_scripts': ['slugify=slugify.__main__:main']},
+    entry_points={'console_scripts': ['slugify=slugify.cli:main']},
 )
diff --git a/slugify/__main__.py b/slugify/__main__.py
index a11989b..4f9e9e1 100644
--- a/slugify/__main__.py
+++ b/slugify/__main__.py
@@ -75,19 +75,3 @@ def slugify_params(args):
         lowercase=args.lowercase,
         replacements=args.replacements
     )
-
-
-def main(argv=None): # pragma: no cover
-    """ Run this program """
-    if argv is None:
-        argv = sys.argv
-    args = parse_args(argv)
-    params = slugify_params(args)
-    try:
-        print(slugify(**params))
-    except KeyboardInterrupt:
-        sys.exit(-1)
-
-
-if __name__ == '__main__': # pragma: no cover
-    main()
diff --git a/slugify/cli.py b/slugify/cli.py
index 0fee3de..8c2dfef 100644
--- a/slugify/cli.py
+++ b/slugify/cli.py
@@ -1,9 +1,14 @@
 import sys
-from slugify import slugify
+import slugify
+import slugify_params
 
-def main(): # pragma: no cover
-    if len(sys.argv) < 2:
-        print("Usage %s TEXT TO SLUGIFY" % sys.argv[0])
-    else:
-        text = ' '.join(sys.argv[1:])
-        print(slugify(text))
+def main(argv=None): # pragma: no cover
+    """ Run this program """
+    if argv is None:
+        argv = sys.argv
+    args = parse_args(argv)
+    params = slugify_params(args)
+    try:
+        print(slugify(**params))
+    except KeyboardInterrupt:
+        sys.exit(-1)
